src/main/java/com/project/bookstore/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,40 @@
+package com.project.bookstore.repository;
+
+import com.project.bookstore.model.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(book);
+            transaction.commit();
+            return book;
+        } catch (RuntimeException e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();

```

Comment:
this will not work correctly

Reply:
Why? Can't find information about it.

Reply:
because EntityManager will be closed before entering catch block
use try-catch-finally

Reply:
And better ask all questions in java chat - it will save your time)